runner/utils.py

```diff
@@ -0,0 +1,286 @@
+        # we wait all the tasks are complete
+        if not incomplete_tasks:
+            return JobStatus(JobState.COMPLETE,
+                             "Job {} successfully completed.".format(job_id))
+        else:
+            logger.info("Job [{}] is running".format(job_id))
+            time.sleep(10)
+
+    return JobStatus(JobState.NOT_COMPLETE,
+                     "ERROR: Tasks did not reach 'Completed' state within timeout period of: " + str(timeout))
+
+
+def check_task_output(batch_service_client: batch.BatchExtensionsClient, job_id: str, expected_file_output_name: str) -> JobStatus:
+    """Prints the stdout.txt file for each task in the job.
+
+    :param batch_service_client: The batch client to use.
+    :type batch_service_client: `Azure.Batch.BatchExtensionsClient`
+    :param job_id: The id of the job with task output files to print.
+    :type job_id: str
+    :param expected_file_output_name: The file name of the expected output
+    :type expected_file_output_name: str
+    :return: The job status, with a message saying the state 
+    :rtype: 'util.JobStatus'
+    """
+
+    tasks = batch_service_client.task.list(job_id)
+
+    for task in tasks:
+        all_files = batch_service_client.file.list_from_task(
+            job_id, task.id, recursive=True)
+
+        for f in all_files:
+            if expected_file_output_name in f.name:
+                logger.info(
+                    "Job [{}] expected output matched {}".format(
+                        job_id, expected_file_output_name))
+                return JobStatus(JobState.COMPLETE,
+                                 "File found {0}".format(expected_file_output_name))
+
+    return JobStatus(JobState.UNEXPECTED_OUTPUT, ValueError(
+        "Error: Cannot find file {} in job {}".format(expected_file_output_name, job_id)))
+
+
+def cleanup_old_resources(blob_client: azureblob.BlockBlobService, days:int=7):
+    """
+    Delete any storage container that has been around for 7 days by default. 
+
+    :param blob_client: A blob service client.
+    :type blob_client: `azure.storage.blob.BlockBlobService`
+    :param days: If the storage account is older than this number of days delete it, default = 7
+    :param int: 
+    """
+    # The current time 7 days ago. 
+    timeout = utc.localize(datetime.datetime.now()) + datetime.timedelta(days=-days)
+
+    try:
+        for container in blob_client.list_containers():
+            if container.properties.last_modified < timeout:
+                if 'fgrp' in container.name:
+                    logger.info(
+                        "Deleting container {}, it is older than 7 days.".format(container.name))
+                    blob_client.delete_container(container.name)
+    except Exception as e:
+        logger.error(e)
+        raise e
+
+
+def create_thread_collection(method_name: str, job_managers: 'list[job_manager.JobManager]', *args):
```

Comment:
:Bulb: please add better explanation of this `Thread` logic here? I am coming from this opinion: https://stackoverflow.com/questions/43747261/when-to-use-threading-and-how-many-threads-to-use 

* Hence in current implementation I might have missed some thing, so add a scenario so that it will help to understand why `threads` are used here. 
* A scenario will help in understand.